titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,401 @@
+    ]
+
+    semaphore = asyncio.Semaphore(int(os.getenv("MOSAIC_CONCURRENCY", 10)))
+
+    values = []
+    with utils.Timer() as t:
+        async with semaphore:
+            for fut in asyncio.as_completed(futures):
+                try:
+                    values.append(await fut)
+                except Exception:
+                    continue
+    if timings:
+        headers["X-Server-Timings"] = "; ".join(
+            ["{} - {:0.2f}".format(name, time * 1000) for (name, time) in timings]
+        )
+
+    return {"coordinates": [lon, lat], "values": values}
+
+
+@router.get(r"/tiles/{z}/{x}/{y}", **tile_response_codes)
+@router.get(r"/tiles/{z}/{x}/{y}\.{format}", **tile_response_codes)
+@router.get(r"/tiles/{z}/{x}/{y}@{scale}x", **tile_response_codes)
+@router.get(r"/tiles/{z}/{x}/{y}@{scale}x\.{format}", **tile_response_codes)
+async def mosaic_tile(
+    z: int = Path(..., ge=0, le=30, description="Mercator tiles's zoom level"),
+    x: int = Path(..., description="Mercator tiles's column"),
+    y: int = Path(..., description="Mercator tiles's row"),
+    scale: int = Query(
+        1, gt=0, lt=4, description="Tile size scale. 1=256x256, 2=512x512..."
+    ),
+    format: ImageType = Query(None, description="Output image type. Default is auto."),
+    pixel_selection: PixelSelectionMethod = Query(
+        PixelSelectionMethod.first, description="Pixel selection method."
+    ),
+    image_params: CommonTileParams = Depends(),
+    mosaic_params: CommonMosaicParams = Depends(),
+):
+    """Read MosaicJSON tile"""
+    # TODO: Maybe use ``read_mosaic`` defined above (depending on cache behavior which is still TBD)
+    pixsel = pixel_selection.method()
+    timings = []
+    headers: Dict[str, str] = {}
+
+    with utils.Timer() as t:
+        with MosaicBackend(mosaic_params.mosaic_path) as mosaic:
+            assets = mosaic.tile(x=x, y=y, z=z)
+            if not assets:
+                raise TileNotFoundError(f"No assets found for tile {z}/{x}/{y}")
+    timings.append(("Read-mosaic", t.elapsed))
+
+    tilesize = 256 * scale
+
+    # Rio-tiler-mosaic uses an external ThreadPoolExecutor to process multiple assets at once but we want to use the
+    # executor provided by the event loop.  Instead of calling ``rio_tiler_mosaic.mosaic.mosaic_tiler`` directly we will
+    # transcribe the code here and use the executor provided by the event loop.  This also means we define this function
+    # as a coroutine (even though nothing that is called is a coroutine), since the event loop's executor isn't
+    # available in normal ``def`` functions.
+    # https://github.com/cogeotiff/rio-tiler-mosaic/blob/master/rio_tiler_mosaic/mosaic.py#L37-L102
+    _tiler = partial(
+        cogeoTiler,
+        tile_x=x,
+        tile_y=y,
+        tile_z=z,
+        tilesize=tilesize,
+        indexes=image_params.indexes,
+        # expression=image_params.expression, # TODO: Figure out why expression kwarg doesn't work
```

Comment:
it's not working because you are using rio-tiler and not rio-tiler-crs (which as a native handle for expression). This might change in near future but for now I'll advice to use rio-tiler-crs (as in the COG endpoint)

Reply:
FYI using rio-tiler-crs you'll need to use a wrapper because of the contextManager

```python
with COGReader(asset) as cog:
    data, mask = cog.tile(x, y, z)
```

Reply:
Yeah I'll probably add a few more wrappers to global scope for now.  Ideally when we dedup code we can use dependency injection to wrap mosaic logic around the `with COGReader` block used by the cog routes.